pokecat/__init__.py

```diff
@@ -345,24 +345,34 @@ def populate_pokeset(pokeset, skip_ev_check=False):
         warn("rarity is %d, which is surprisingly high. Note that 1.0 is the default "
              "and high values mean the Pokémon gets chosen more often." % rarity)
 
-    # fix default biddable value
-    if pokeset["biddable"] is None:
-        pokeset["biddable"] = not pokeset["shiny"]
-    if not isinstance(pokeset["biddable"], bool):
-        raise ValueError("biddable must be a boolean (true or false), not %s" % type(pokeset["biddable"]))
-
-    # fix default hidden value
-    if pokeset["hidden"] is None:
-        pokeset["hidden"] = pokeset["shiny"]
-    if not isinstance(pokeset["hidden"], bool):
-        raise ValueError("hidden must be a boolean (true or false), not %s" % type(pokeset["hidden"]))
-
-    if pokeset["biddable"] and pokeset["hidden"]:
-        warn("Set is biddable, but also hidden, which doesn't make sense.")
-    if pokeset["shiny"] and pokeset["biddable"] and Suppressions.PUBLIC_SHINY not in suppressions:
-        warn("Set is shiny, but also biddable, which means it can be used in token matches. Is this intended?")
-    if pokeset["shiny"] and not pokeset["hidden"] and Suppressions.PUBLIC_SHINY not in suppressions:
-        warn("Set is shiny, but not hidden, which means it is publicly visible. Is this intended?")
+    if pokeset["shiny"] and Suppressions.PUBLIC_SHINY not in suppressions:
+        if pokeset["biddable"] is True:
+            warn("Set is shiny, but also biddable, which means it can be used in token "
+                 "matches. Is this intended?")
+        if pokeset["hidden"] is False:
+            warn("Set is shiny, but not hidden, which means it is publicly visible. Is "
+                 "this intended?")
+
+    if pokeset["biddable"] is pokeset["hidden"] is True:
+        warn("Set is explicitly biddable, but also explicitly hidden, which doesn't make sense.")
+    else:
+        # fix default biddable value
+        if pokeset["biddable"] is None:
```

Comment:
if the above warnings emits, this codepath isn't executed, meaning `biddable` and `hidden` may still be None after population. Is this expected behaviour?

Reply:
also the fields aren't being validated either. The above should be an error, or cause sensible defaults I guess